src/unitxt/inference.py

```diff
@@ -142,30 +159,42 @@ def prepare(self):
         credentials = Credentials(api_key=api_key)
         self.client = Client(credentials=credentials)
 
+        if (
+            getattr(self.parameters, "type", None)
+            == "ibm_gen_ai_inference_engine_params"
+        ):
+            logger.warning(
+                "Artifact IbmGenAiInferenceEngineParams is deprecated for GenAI parameter definition, please use plain "
+                "dictionary. This will raise an exception in the future"
+            )
+            self.parameters = self.parameters.to_dict()
+        TextGenerationParameters.model_validate(self.parameters)
+
     def _infer(self, dataset):
-        from genai.schema import TextGenerationParameters
+        from tqdm import tqdm
 
-        genai_params = TextGenerationParameters(
-            max_new_tokens=self.parameters.max_new_tokens,
-            min_new_tokens=self.parameters.min_new_tokens,
-            random_seed=self.parameters.random_seed,
-            repetition_penalty=self.parameters.repetition_penalty,
-            stop_sequences=self.parameters.stop_sequences,
-            temperature=self.parameters.temperature,
-            top_p=self.parameters.top_p,
-            top_k=self.parameters.top_k,
-            typical_p=self.parameters.typical_p,
-            decoding_method=self.parameters.decoding_method,
-        )
+        prompts = [instance["source"] for instance in dataset]
 
-        return [
-            response.results[0].generated_text
-            for response in self.client.text.generation.create(
-                model_id=self.model_name,
-                inputs=[instance["source"] for instance in dataset],
-                parameters=genai_params,
-            )
-        ]
+        pb = tqdm(desc="Running text generation", total=len(prompts))
+        predictions = []
+        metadata = []
+
+        for resp in self.client.text.generation.create(
+            inputs=prompts, model_id=self.model_name, parameters=self.parameters
+        ):
+            for result in resp.results:
+                predictions.append(result.generated_text)
+                metadata.append(
+                    {
+                        "meta": resp.model_dump(exclude={"results"}, exclude_none=True),
```

Comment:
@jezekra1 - what do you need the meta data for? If each inference engine, returns different meta data, then the code that uses it, can not do much with it except record it.